packages/core/src/com/communication.ts

```diff
@@ -252,6 +258,7 @@ export class Communication {
                     data: { api, method, args },
                     callbackId,
                     origin,
+                    forwardingChain: [this.rootEnvId],
```

Comment:
why call needs  `forwardingChain: [this.rootEnvId]` when it's direct call?

Reply:
after some thinking process i think `addOrRemoveListener`, `callMethod`, `forwardUnlisten` and `forwardListenmessage` should accept `forwardingChain` passed by caller

Reply:
this way `forwardingChain` will be populated only inside `forwardMessage` and all functions mentioned above will respect and pass through `forwardingChain` value

Reply:
thanks, that really make sense. updated